File: Npc/11001059.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11001059: Danak
/// </summary>
public class _11001059 : NpcScript {
    internal _11001059(INpcScriptContext context) : base(context) {
        Id = 30;
    }

    public override bool Next(int selection = 0) {
        switch (Id) {
            case 0:
                // $script:0831180407003617$ 
                // - Gasp! Did my wife send you?
                return true;
            case 30:
                // $script:0831180407003620$ 
                // - No one understands the burning passion of this lonely man... Sigh... 
                return true;
            default:
                return true;
        }
    }
}

File: Npc/11001059.py
""" 11001059: Danak """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 30

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0831180407003617$
        # - Gasp! Did my wife send you?
        return None # TODO

    def __30(self, pick: int) -> int:
        # $script:0831180407003620$
        # - No one understands the burning passion of this lonely man... Sigh... 
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (30, 0):
            return Option.CLOSE
        return Option.NONE
